Snowflake.Data/Core/SFSession.cs

```diff
@@ -149,21 +172,25 @@ internal async Task OpenAsync(CancellationToken cancellationToken)
 
         internal void close()
         {
-            var queryParams = new Dictionary<string, string>();
-            queryParams[RestParams.SF_QUERY_SESSION_DELETE] = "true";
-            queryParams[RestParams.SF_QUERY_REQUEST_ID] = Guid.NewGuid().ToString();
-            queryParams[RestParams.SF_QUERY_REQUEST_GUID] = Guid.NewGuid().ToString();
-
-            SFRestRequest closeSessionRequest = new SFRestRequest
-            {
-                Url = BuildUri(RestPath.SF_SESSION_PATH, queryParams),
-                authorizationToken = string.Format(SF_AUTHORIZATION_SNOWFLAKE_FMT, sessionToken)
-            };
-          
-            var response = restRequester.Post<CloseResponse>(closeSessionRequest);
-            if (!response.success)
+            // Only close the session if it is opened
+            if (null != sessionToken)

```

Comment:
Maybe change this to 
```
if (sessionToken == null) return;

....
```
This way, you don't need to include majority of the code in this if block.